components/x-audio/src/redux/swipe-down-to-minimise.js

```diff
@@ -0,0 +1,31 @@
+const EXPANDED_PLAYER_HEIGHT = 360;
+const DISPLAY_EXPANDED_THRESHOLD = EXPANDED_PLAYER_HEIGHT / 2;
+
+function setPositionY (target, posY) {
+	target.style.transform = `translate3d(0, ${posY}px, 0)`;
+}
+
+export default (event, playerActions) => {
+
+	const direction = event.offsetDirection;
+	const directionDown = direction === 16;
+
+	if (!directionDown) {
+		return;
+	}
+
+	const expandedPlayer = event.target.closest("#audio-player-expanded");
+	const posY = event.deltaY;
+
+	setPositionY(expandedPlayer, posY);
+
+	// SWIPE ENDED
+	if (event.isFinal) {
+		if (posY >= DISPLAY_EXPANDED_THRESHOLD) {
+			playerActions.onMinimise();
+		}
+
+		setPositionY(expandedPlayer, 0);
+	}
+
+}
```

Comment:
Wondering if the `redux` directory is the right location for this file. What do you think?

Reply:
@benbarnett Thank you for this comment!! I felt wrong when I put the file there, however I was not sure where it should be. Could you suggest where is a good place? `x-audio/src/lib`?

Reply:
📝 I moved this file in components folder where other js file and helper file are. We will clean up when everything finished.